fsspec/spec.py

```diff
@@ -750,25 +755,35 @@ def get(self, rpath, lpath, recursive=False, **kwargs):
         """
         from .implementations.local import make_path_posix
 
+        callback = as_callback(kwargs.pop("callback", None))
         if isinstance(lpath, str):
             lpath = make_path_posix(lpath)
         rpaths = self.expand_path(rpath, recursive=recursive)
         lpaths = other_paths(rpaths, lpath)
-        for lpath, rpath in zip(lpaths, rpaths):
+
+        callback.lazy_call("set_size", len, lpaths)
+        for lpath, rpath in callback.wrap(zip(lpaths, rpaths)):
+            branch(callback, rpath, lpath, kwargs)
             self.get_file(rpath, lpath, **kwargs)
 
     def put_file(self, lpath, rpath, **kwargs):
         """Copy single file to remote"""
         if os.path.isdir(lpath):
             self.makedirs(rpath, exist_ok=True)
-        else:
-            with open(lpath, "rb") as f1:
-                self.mkdirs(os.path.dirname(rpath), exist_ok=True)
-                with self.open(rpath, "wb", **kwargs) as f2:
-                    data = True
-                    while data:
-                        data = f1.read(self.blocksize)
-                        f2.write(data)
+            return None
+
+        callback = as_callback(kwargs.pop("callback", None))
+        with open(lpath, "rb") as f1:
+            callback.lazy_call(
+                "set_size", lambda: os.stat(lpath, follow_symlinks=False).st_size
```

Comment:
```
size = f1.seek(0, 2)
f1.seek(0)
```
seems marginally faster (time was already spent opening the file).

Reply:
Makes sense, considering we can seek on these sort of local files.